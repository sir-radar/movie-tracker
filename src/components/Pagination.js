import React from 'react';

const Paginattion = ({searchMore, page, pages}) => {

  const nextPage = (pageNumber) =>{
    searchMore(pageNumber)
  }

  return (
    <nav className="w-100 pr-3 mt-3">
      <ul className="pagination float-right">
        {
          page > 1 ? <li className="page-item option" onClick={() => nextPage(--page)}>
                        <span className="page-link">Previous</span>
                     </li>
                : <li className="page-item disabled">
                    <span className="page-link">Previous</span>
                  </li>
        }
        <li className="page-item">
            <span className="page-link">{`Page ${page} of ${pages}`}</span>
        </li>
        {
          page < pages ? <li className="page-item option" onClick={() => nextPage(++page)}>
                                <span className="page-link">Next</span>
                             </li>
                  : <li className="page-item disabled">
                      <span className="page-link">Next</span>
                    </li>
        }
        
      </ul>
    </nav>
  );
}

export default Paginattion;
