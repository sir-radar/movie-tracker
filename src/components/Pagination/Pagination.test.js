import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Pagination from './Pagination';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

afterEach(cleanup)

const renderComponent = (props) => {
  const div = document.createElement("div");
  return render(<Pagination {...props}/>, div)
}

const propsData = {
  page: 1,
  pages: 5, 
  searchMore: jest.fn()
};

it("renders without crashing", () => {
  renderComponent(propsData)
})

it("renders pagination correctly", () => {
  const { queryByTestId } = renderComponent(propsData)
  expect(queryByTestId('active-prev')).toBeNull()
  expect(queryByTestId('inactive-prev')).not.toBeNull()
  expect(queryByTestId('pagination-detail')).toHaveTextContent(`Page ${propsData.page} of ${propsData.pages}`)
  expect(queryByTestId('active-next')).not.toBeNull()
  expect(queryByTestId('inactive-next')).toBeNull()
})

it("renders active previous button if page is greater than one", () => {
  const newpage = 2;
  const { queryByTestId } = renderComponent({...propsData, page: newpage})
  expect(queryByTestId('active-prev')).not.toBeNull()
  expect(queryByTestId('inactive-prev')).toBeNull()
  expect(queryByTestId('pagination-detail')).toHaveTextContent(`Page ${newpage} of ${propsData.pages}`)
  expect(queryByTestId('active-next')).not.toBeNull()
  expect(queryByTestId('inactive-next')).toBeNull()
})

it("renders inactive next button if page is greater or equal to pages", () => {
  const newpage = 5;
  const { queryByTestId } = renderComponent({...propsData, page: newpage})
  expect(queryByTestId('active-prev')).not.toBeNull()
  expect(queryByTestId('inactive-prev')).toBeNull()
  expect(queryByTestId('pagination-detail')).toHaveTextContent(`Page ${newpage} of ${propsData.pages}`)
  expect(queryByTestId('active-next')).toBeNull()
  expect(queryByTestId('inactive-next')).not.toBeNull()
})

it("should call searchMore if active previous button or active next button is clicked", () => {
  const newpage = 2;
  const { queryByTestId } = renderComponent({...propsData, page: newpage})
  expect(queryByTestId('active-prev')).not.toBeNull()
  fireEvent.click(queryByTestId('active-prev'));
  expect(propsData.searchMore).toHaveBeenCalled()
  expect(queryByTestId('inactive-prev')).toBeNull()
  expect(queryByTestId('pagination-detail')).toHaveTextContent(`Page ${newpage} of ${propsData.pages}`)
  expect(queryByTestId('active-next')).not.toBeNull()
  fireEvent.click(queryByTestId('active-next'));
  expect(propsData.searchMore).toHaveBeenCalled()
  expect(queryByTestId('inactive-next')).toBeNull()
})

it("matches snapshot", () => {
  const tree = renderer.create(<Pagination {...propsData}/>).toJSON();
  expect(tree).toMatchSnapshot();
})