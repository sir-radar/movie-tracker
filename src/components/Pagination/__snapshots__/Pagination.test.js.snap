// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="w-100 pr-3 mt-3"
  data-testid="pagination"
>
  <ul
    className="pagination float-right"
  >
    <li
      className="page-item disabled"
      data-testid="inactive-prev"
    >
      <span
        className="page-link"
      >
        Previous
      </span>
    </li>
    <li
      className="page-item"
      data-testid="pagination-detail"
    >
      <span
        className="page-link"
      >
        Page 1 of 5
      </span>
    </li>
    <li
      className="page-item option"
      data-testid="active-next"
      onClick={[Function]}
    >
      <span
        className="page-link"
      >
        Next
      </span>
    </li>
  </ul>
</nav>
`;
